stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/BaseMessageItemViewHolder.kt

```diff
@@ -1,19 +1,33 @@
 package io.getstream.chat.android.ui.messages.adapter
 
 import android.view.View
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
 import androidx.recyclerview.widget.RecyclerView
 import com.getstream.sdk.chat.adapter.MessageListItem
 import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.BackgroundDecorator
 import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.Decorator
 import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.GapDecorator
+import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.MaxPossibleWidthDecorator
 
 public abstract class BaseMessageItemViewHolder<T : MessageListItem>(itemView: View) : RecyclerView.ViewHolder(itemView) {
-    private val decorators = listOf<Decorator>(BackgroundDecorator(), GapDecorator())
+    private val decorators = listOf<Decorator>(BackgroundDecorator(), GapDecorator(), MaxPossibleWidthDecorator())
 
     public fun bind(data: T) {
         decorators.forEach { it.decorate(this, data) }
         bindData(data)
     }
 
     public abstract fun bindData(data: T)
+
+    protected fun constraintView(isMine: Boolean, view: View, layout: ConstraintLayout) {
+        ConstraintSet().apply {
+            clone(layout)
+            clear(view.id, ConstraintSet.LEFT)
+            clear(view.id, ConstraintSet.RIGHT)
+            val anchorSide = if (isMine) ConstraintSet.END else ConstraintSet.START
```

Comment:
When message `isMine()==true` the `anchorSide` is set to the `right` for the left-to-right type alphabets and to the `left` for right-to-left alphabets (Arabic, Hebrew). Probably that's fine. However, if we don't want to support different alphabets types now it may be better to use `LEFT` and `RIGHT` instead of `END` and `START`.

Reply:
You're right. right-to-left support and other kinds of customization will be brought in the future